### PR TITLE
[TS] LPS-172417

### PR DESCRIPTION
**Pull request by ChrisKian**
https://issues.liferay.com/browse/LPS-172417

When configured with a reverse proxy and javascript/CSS fasload set to true, some resources are not properly retrieved in the combo servlet due to how the resource name is assembled.

In this situation, a resource name will look something like this:
`com_liferay_server_admin_web_portlet_ServerAdminPortlet:/proxyPath/o/...`

Because we need to remove the proxy path and context path, but the modulePortletId comes first, the replacing logic won't work, and our combo files will be incomplete, leading to unexpected errors in the portal.

My first commit also resolves a small issue where the proxy path is added twice to the DirectServletRegistry path.

Please let me know if you have any questions, thanks!

### CODE REVIEW

**Comment by Kvale1733** [comment]
@zhangyan0701 I think this issue is a duplicate of: [LPS-170627](https://issues.liferay.com/browse/LPS-170627). Looks like you caught case 5. I didn't find any regressions but I'm having difficulty testing the issue locally. I tried running the test you created for that but I'm missing the directory for that conf file and the I can't get the portal to startup following the steps for this ticket. Can you help me verify this fix?

**Comment by zhangyan0701** [comment]
@Kvale1733 Yes, the case 5 is the same issue as the one I reported. And with the changes the issue can be fixed. I also tested other cases in LPS-172417, all are fixed. And I didn't find regressions neither.

File: portal-impl/src/com/liferay/portal/servlet/DirectServletRegistryImpl.java [hunk 1]
```diff
@@ -223,8 +223,7 @@ protected void updateFileLastModified(String path, Servlet servlet) {
 	private static class PathModulePrefixHolder {
 
 		private static final String _PATH_MODULE_PREFIX = StringBundler.concat(
-			PortalUtil.getPathProxy(), PortalUtil.getPathContext(),
-			Portal.PATH_MODULE, StringPool.SLASH);
+			PortalUtil.getPathContext(), Portal.PATH_MODULE, StringPool.SLASH);
```
Comment (tinatian) [inline portal-impl/src/com/liferay/portal/servlet/DirectServletRegistryImpl.java:226]: @ChrisKian pathProxy was added to fix https://issues.liferay.com/browse/LPS-108204, why do you think it is safe to remove?

Comment (tinatian) [reply portal-impl/src/com/liferay/portal/servlet/DirectServletRegistryImpl.java:226]: @ChrisKian I see your point, it is safe to remove

**Comment by tinatian** [comment]
@zhangyan0701 Could you please verify if https://issues.liferay.com/browse/LPS-108204 still fixed?

Thanks.
Tina

**Comment by zhangyan0701** [comment]
Hi @tinatian @ChrisKian 

Ticket [LPS-108204](https://issues.liferay.com/browse/LPS-108204) is still fixed will the changes here.

But the UI issues in [LPS-172417](https://issues.liferay.com/browse/LPS-172417) become reproducible with non-root context (with the configurations in [LPS-108204](https://issues.liferay.com/browse/LPS-108204)), this should be caused by this pull, I didn't reproduce on master head.

**Comment by ChrisKian** [comment]
I will try to resolve this issue next week.

**Comment by ChrisKian** [comment]
Resent here: https://github.com/liferay-core-infra/liferay-portal/pull/3106

**Pull request closed without merging**